WordPress/gutenberg
# Fix trailing spaces in navigation block classnames

## What?
Fixes the very minor issue in #67999, adds some trimming to navigation block classnames

## Testing Instructions
1. View a navigation block on the frontend of a site, ensure it's one that's configured to show the Overlay Menu on mobile or always.
2. Open dev tools and either reduce the width of the browser or use the mobile device simulation thing
3. Inspect the navigation hamburger menu (or two buns with no burger) button
4. Check that the `wp-block-navigation__responsive-container-open` class on the button has no trailing space
5. Click the hamburger button
6. Check that the `wp-block-navigation__responsive-container` doesn't have two spaces after it.

## Screenshots or screencast 



|Before|After|
|-|-|
| <img width="433" alt="Screenshot 2024-12-20 at 10 01 03 am" src="https://github.com/user-attachments/assets/1453346e-5b97-4f40-bd0f-36bcde917d45" /> | <img width="430" alt="Screenshot 2024-12-20 at 10 00 40 am" src="https://github.com/user-attachments/assets/2af8f197-40f5-485a-a034-5caf8630f0b4" /> |
